src/main/java/org/group23/ControllerStructure.java

```diff
@@ -1,55 +1,108 @@
 package org.group23;
 
-
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.ui.Model;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
+import org.springframework.web.bind.annotation.*;
 
 @org.springframework.stereotype.Controller
 public class ControllerStructure {
 
     @Autowired
     QuestionRepository questionRepository;
-
     @Autowired
     SurveyRepository surveyRepository;
 
-    @GetMapping("/hello")
-    public String sayHello(Model model) {
-        model.addAttribute("message", "Hello, World!");
-        return "helloPage";
+    @GetMapping("/createSurvey")
+    public String createSurveyForm(Model model) {
+        model.addAttribute("survey", new Survey());
+        return "createSurvey";
+    }
+
+    @PostMapping("/saveSurveyName")
+    public String saveSurveyName(@ModelAttribute("survey") Survey survey) {
+        surveyRepository.save(survey);
+        return "redirect:/addRemoveQuestions/" + survey.getId();
     }
 
+    @GetMapping("/addRemoveQuestions/{surveyId}")
+    public String addRemoveQuestionsForm(@PathVariable Long surveyId, Model model) {
+        // Get the survey by ID
+        Survey survey = surveyRepository.findById(surveyId).orElse(null);
 
-    //These methods were for testing CrudRepositories and should definitely be changed ASAP :)
+        if (survey != null) {
+            model.addAttribute("survey", survey);
+            return "addRemoveQuestions";
+        } else {
+            // Handle survey not found
+            return "error";
+        }
+    }
 
+    @PostMapping("/addQuestion/{surveyId}")
+    public String addQuestion(
+            @PathVariable Long surveyId,
+            @ModelAttribute("survey") Survey survey,
+            @RequestParam String questionText,
```

Comment:
Validate length before adding to database!

Reply:
![image](https://github.com/Andrei486/sysc-4806-project/assets/55627866/d3fcb6df-2d20-4ccd-8a20-de2172715e79)

Reply:
done